### ISSUE TITLE
Entering an invalid value for SSL when connecting to postgres crashes evidence

### ISSUE THREAD

**Issue by scsmithr** [evidence-dev/evidence#1228]
## Steps To Reproduce

I'm using the template repo here: https://github.com/evidence-dev/template

1. `npm install` & `npm run dev`
2. Go to `127.0.0.1:3000/settings`
3. Select postgres, enter in some information, then enter in an invalid value for the `SSL` field (I entered "require" thinking it mapped to `sslmode`).
4. Observe network error and the following logs:

```
  Connection Test running.../Users/sean/Code/github.com/glaredb/reports/node_modules/pg/lib/connection.js:85
        if ('key' in self.ssl) {
                  ^

TypeError: Cannot use 'in' operator to search for 'key' in require
    at Socket.<anonymous> (/Users/sean/Code/github.com/glaredb/reports/node_modules/pg/lib/connection.js:85:19)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
```

I am also unable to start up evidence again as I'm met with the following:

```
% npm run dev

> my-evidence-project@0.0.1 dev
> evidence dev --open /


Forced re-optimization of dependencies

  VITE v4.3.9  ready in 481 ms

  ➜  Local:   http://127.0.0.1:3000/
  ➜  Network: use --host to expose
  ➜  press h to show help
  orders_by_month running.../Users/sean/Code/github.com/glaredb/reports/node_modules/pg/lib/connection.js:85
        if ('key' in self.ssl) {
                  ^

TypeError: Cannot use 'in' operator to search for 'key' in require
    at Socket.<anonymous> (/Users/sean/Code/github.com/glaredb/reports/node_modules/pg/lib/connection.js:85:19)
    at Object.onceWrapper (node:events:628:26)
    at Socket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Socket.Readable.push (node:internal/streams/readable:228:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
```

I removed the node modules and tried reinstalling, but still get the above. It seems like my invalid db configuration is sticking around, and I don't know where or how it's set (are settings persisted to a config file somewhere?).

### Environment

- Node version (`node -v`): v16.20.2
- npm version (`npm -v`): 8.19.4
- OS: macos
- Browser: firefox
- Database: postgres (well... trying to connect to glaredb using pg protocol)

## Expected Behavior

Have evidence not exit on invalid value for 'SSL' field. Also have a way of understanding where these values are persisted/being read from so that I can manually remove the invalid configuration.

## Actual Behaviour

See above errors.

## Workarounds

Quick fix for me would be to know how I can reset my configuration. I don't have any `EVIDENCE_*` env vars set, so I'm not sure how evendence is reading the invalid value.

**Comment by scsmithr** [comment]
Figured out where to go for the settings configuration. Saw this in cloud:

> Upload your evidence.settings.json file.
The file should be inside your project in the hidden folder  .evidence/template

I was looking at the docs previously and was not finding anything other than the environment vars.

**Comment by archiewood** [comment]
This is an annoying state to reach, it's a bit of a trap.

I'm thinking there should be something like a "Evidence: Clear configurations and cache" kind of command in the VSCode extension that deletes almost everything in the .evidence/template folder, which would:
- Purge the query cache
- Purge the stored credentials
- Purge any Svelte kit generated files

as this can normally rescue you from most situations

**Comment by archiewood** [comment]
Separately, feels like we should do something to error handle the SSL inputs so that you can't get trapped

**Comment by scsmithr** [comment]
I think a "Clear ..." command in the vscode extension could be useful, but I'm also not using the extension (and so that wouldn't have helped me here).

What would have helped me:
- Include a mention of `evidence.settings.json` in the docs site. I looked here first and searched "settings" to try to find something about where configuration is stored, and only found mentions of environment vars.
- Include something in the template repo readme about where settings are saved.

I only learned about `evidence.settings.json` after checking the credentials stuff in evidence.app and seeing a snippet about it.
